screens/LoginScreen.tsx

```diff
@@ -1,125 +1,98 @@
-            <Text style={authStyles.title}>Log in</Text>
-            <TextInput
-              style={authStyles.input}
-              mode="outlined"
-              label="Username"
-              value={form.username}
-              onChangeText={(e) => setForm({ ...form, username: e })}
-            />
-            <TextInput
-              style={authStyles.input}
-              mode="outlined"
-              label="Password"
-              secureTextEntry={!showPassword}
-              right={
-                <TextInput.Icon
-                  icon={showPassword ? 'eye-off' : 'eye'}
-                  onPress={() => setShowPassword(!showPassword)}
-                />
-              }
-              value={form.password}
-              onChangeText={(e) => setForm({ ...form, password: e })}
+    <>
+      <SafeAreaView
+        style={{ backgroundColor: colors.primaryContainer, flex: 1 }}
+      >
+        <ScrollView keyboardShouldPersistTaps={'handled'}>
+          <View style={authStyles.root}>
+            <Image
+              source={hushallet_logo}
+              resizeMode="contain"
+              style={authStyles.logo}
             />
-            <Button
-              style={authStyles.button}
-              icon="login"
-              mode="contained"
-              onPress={() => navigation.replace('HomeNavigator')}
-            >
-              Log In
-            </Button>
-            <View style={authStyles.linkTextContainer}>
-              <Text>Dont have an account? </Text>
-              <TouchableOpacity onPress={() => navigation.navigate('Register')}>
-                <Text style={authStyles.linkText}>Sign up</Text>
-              </TouchableOpacity>
+            <View style={authStyles.container}>
+              <Text style={authStyles.title}>Log in</Text>
+              {error && (
+                <View>
+                  <Text style={{ color: colors.error }}>{error}</Text>
+                </View>
+              )}
+              <TextInput
+                style={authStyles.input}
+                mode="outlined"
+                label="Username"
+                value={form.username}
+                onChangeText={(e) => setForm({ ...form, username: e })}
+              />
+              <TextInput
+                style={authStyles.input}
+                mode="outlined"
+                label="Password"
+                secureTextEntry={!showPassword}
+                right={
+                  <TextInput.Icon
+                    icon={showPassword ? 'eye-off' : 'eye'}
+                    onPress={() => setShowPassword(!showPassword)}
+                  />
+                }
+                value={form.password}
+                onChangeText={(e) => setForm({ ...form, password: e })}
+              />
+              <Button
+                style={authStyles.button}
+                icon="login"
+                mode="contained"
+                onPress={() => handleLogin()}
```

Comment:
No need to create a "new" function since it doesn't take any parameters, just need to reference it
```suggestion
                onPress={handleLogin}
```